node/src/git.rs

```diff
@@ -34,6 +34,35 @@ pub enum ListRefsError {
     InvalidRef(#[from] RefError),
 }
 
+pub mod refs {
+    use super::*;
+
+    // Where project information is kept.
+    pub const IDENTITY_BRANCH: &str = "radicle/id";
+
+    pub mod storage {
+        use super::*;
+
+        pub fn peer_branch(remote: &RemoteId, branch: &str) -> String {
+            format!("refs/remotes/{remote}/heads/{branch}")
+        }
+
+        pub fn peer_id(remote: &RemoteId) -> String {
+            peer_branch(remote, IDENTITY_BRANCH)
+        }
+    }
+
+    pub mod workdir {
+        pub fn local_branch(branch: &str) -> String {
+            format!("refs/heads/{branch}")
+        }
+
+        pub fn remote_branch(remote: &str, branch: &str) -> String {
+            format!("refs/remotes/{remote}/{branch}")
+        }
+    }
+}
```

Comment:
Using RefLike will require more elaborate changes, one being error handling.  Since these changes were about defining the set of common refs, it may be best to leave introducing these types in another PR.

Reply:
> Using RefLike will require more elaborate changes, one being error handling. Since these changes were about defining the set of common refs, it may be best to leave introducing these types in another PR.

I don't think you'd need error handling. Also, note that @cloudhead said `RefString` and not `RefLike` -- two different iterations of the same concept :) 

You could model the above as:
```rust
/// Create a remote branch under the `remote` with the name `branch`.
pub fn remote_branch<'a>(remote: Component<'a>, branch: RefString) -> RefString {
    // into_inner turns the component into a Cow<'a, RefStr>, which we can `join` with `branch`
    let tail = remote.into_inner().join(branch);
    refs_remotes(tail)
}
```

However, there's no good way of creating a `Component` yet -- other than known ones. I think we could add a `component!` macro that makes it easier.

Reply:
Just wanted to reply to this because I actually missed that there is a `component!` macro :)

https://github.com/radicle-dev/radicle-link/blob/master/git-ref-format/macro/src/lib.rs#L49-L55